HLTrigger/Egamma/plugins/HLTEgammaCombMassFilter.cc

```diff
@@ -35,6 +36,7 @@ void HLTEgammaCombMassFilter::fillDescriptions(edm::ConfigurationDescriptions& d
   makeHLTFilterDescription(desc);
   desc.add<edm::InputTag>("firstLegLastFilter", edm::InputTag("firstFilter"));
   desc.add<edm::InputTag>("secondLegLastFilter", edm::InputTag("secondFilter"));
+  desc.add<edm::InputTag>("l1EGCand", edm::InputTag("hltL1IsoRecoEcalCandidate"));
```

Comment:
Since the default value of the tag is not realistic, and the PR has no changes to the python configs, I would naively expect some test to fail because of a misconfigured InputTag. Or not ?

Reply:
scram b runtests passed without failures - if there are python configs where this parameter should be added, they are not used in testing. Are the configs where the module is called such as  HLTrigger/Configuration/test/OnLine_HLT_FULL.py are expected to need changes?

Reply:
fillDescriptions inserts values/parameters if they are missing in the py config, so no changes needed there.
However, if the default of a new parameter is not realistic, then it is either not used or misused. Please check!

Reply:
Hi @Martin-Grunewald , since it passes tests its unclear to me what is meant by the current value is unrealistic, could you clarify?
BTW if I remove that line (L39), when doing HLT emulation we would have error as below:

> ----- Begin Fatal Exception 17-Jun-2024 07:35:03 CEST-----------------------
> An exception of category 'Configuration' occurred while
>    [0] Constructing the EventProcessor
>    [1] Constructing module: class=HLTEgammaCombMassFilter label='hltDoubleEle10Mass50PPRefFilter'
> Exception Message:
> MissingParameter: Parameter 'l1EGCand' not found.
> ----- End Fatal Exception -------------------------------------------------

And if I also remove L27 and L49-51, nothing will be stored in the hltobject. So I think we should keep it as it is?

Reply:
I think the issue @missirol is referring to is the observation that ```hltL1IsoRecoEcalCandidate``` is not part of any recent menu, so there shouldn't be an object with that label in the Event. So it is weird!
Check whether/how ```l1EGCand``` is used.

Reply:
In line 50 you use l1EGTag, not l1EGCand.

Reply:
> In line 50 you use l1EGTag, not l1EGCand.

In line 27,  `l1EGCand` is passed to `l1EGTag_`. And in other filters such as `HLTGenericFilter` and `HLTElectronPixelMatchFilter` we can also see `hltL1IsoRecoEcalCandidate` as the InputTag of l1EGCand.

Reply:
Above you set the value to be used for this new parameter, in case the config does not contain it (yet) - which is the case for this specific plugin. For the othe rplugins you mention whihc also have this parametrer, its valie is specified in the python config, for example:
```
process.hltDiMu5Ele3CaloIdLTrackIdLElectronlegPixelMatchFilter = cms.EDFilter( \
"HLTElectronPixelMatchFilter",
    saveTags = cms.bool( True ),
    candTag = cms.InputTag( "hltDiMu5Ele3CaloIdLTrackIdLElectronlegHEFilter" ),
    l1PixelSeedsTag = cms.InputTag( "hltEgammaElectronPixelSeeds" ),
    npixelmatchcut = cms.double( 1.0 ),
    ncandcut = cms.int32( 2 ),
    l1EGCand = cms.InputTag( "hltEgammaCandidates" ),
    s_a_phi1B = cms.double( 0.0069 ),
    s_a_phi1I = cms.double( 0.0088 ),
    s_a_phi1F = cms.double( 0.0076 ),
    s_a_phi2B = cms.double( 3.7E-4 ),
    s_a_phi2I = cms.double( 7.0E-4 ),
    s_a_phi2F = cms.double( 0.00906 ),
    s_a_zB = cms.double( 0.012 ),
    s_a_rI = cms.double( 0.027 ),
    s_a_rF = cms.double( 0.04 ),
    s2_threshold = cms.double( 0.4 ),
    tanhSO10BarrelThres = cms.double( 0.35 ),
    tanhSO10InterThres = cms.double( 1.0 ),
    tanhSO10ForwardThres = cms.double( 1.0 ),
    useS = cms.bool( False ),
    pixelVeto = cms.bool( False )
)
```

Now, in this PR, the value used is the one from fillDescriptions, which does not exist as a producer in the menu!

Reply:
@pinchunchou 

> So there is a problem!

To add to this, I have a question: even if the menu does run as is, have you checked that you can access the refs to these trigger objects from `hltTriggerSummaryRAW` (`TriggerEventWithRefs`) ? I might be wrong, but I would think one would not be able to access the correct objects if the InputTag which was specified was incorrect (like in this case).

Reply:
@missirol @Martin-Grunewald I see, thanks for the explanation. I have replaced the `hltL1IsoRecoEcalCandidate` to `hltEgammaCandidates` in my newest commit (squashed).

Reply:
Thanks. I see 5 instances of `HLTEgammaCombMassFilter` in the HLT menus.
https://raw.githubusercontent.com/cms-sw/cmssw/9030bf6e5bc33e617ca03ac40e8586ea8b86abc2/HLTrigger/Configuration/test/OnLine_HLT_FULL.py
```
hltDoubleEle8Mass8Filter
hltDoubleEle10Mass50PPOnAAFilter
hltDoubleEle15Mass50PPOnAAFilter
hltDoubleEle10Mass50PPRefFilter
hltDoubleEle15Mass50PPRefFilter
```
Could you please provide the correct value of `l1EGCand` for each of these 5 modules ? Unless it's always `hltEgammaCandidates`, maybe the modules requiring a non-default value should be modified in this PR via a customisation function ([example](https://github.com/cms-sw/cmssw/blob/9030bf6e5bc33e617ca03ac40e8586ea8b86abc2/HLTrigger/Configuration/python/customizeHLTforCMSSW.py#L328)).

Reply:
Yeah I think the correct value of `l1EGCand` for each of these 5 modules should always be `hltEgammaCandidates`

Reply:
> Yeah I think the correct value of `l1EGCand` for each of these 5 modules should always be `hltEgammaCandidates`

I doubt it, since `hltDoubleEle10Mass50PPOnAAFilter` and `hltDoubleEle15Mass50PPOnAAFilter` only exist in the `HIon` HLT menu, and `hltEgammaCandidates` does not exist in the latter. Those two modules probably need to use `hltEgammaCandidatesPPOnAA`.

Reply:
Yes you are correct. In the modules for `HIon` HLT menu we should use `hltEgammaCandidatesPPOnAA`. I looked into the example you provided but not quite sure how to do the customisation for them, could you advice?

Reply:
It could be something as follow (did not test it).
```diff
diff --git a/HLTrigger/Configuration/python/customizeHLTforCMSSW.py b/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
index 4f116461823..0b027666ccc 100644
--- a/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
+++ b/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
@@ -261,6 +261,13 @@ def checkHLTfor43774(process):
 
     return process
     
+def customizeHLTfor44746(process):
+    for modLabel in ['hltDoubleEle10Mass50PPOnAAFilter', 'hltDoubleEle15Mass50PPOnAAFilter']:
+        if hasattr(process, modLabel):
+            mod = getattr(process, modLabel)
+            mod.l1EGCand = cms.InputTag('hltEgammaCandidatesPPOnAA')
+    return process
+
 # CMSSW version specific customizations
 def customizeHLTforCMSSW(process, menuType="GRun"):
 
@@ -270,5 +277,6 @@ def customizeHLTforCMSSW(process, menuType="GRun"):
     # process = customiseFor12718(process)
 
     process = checkHLTfor43774(process)
+    process = customizeHLTfor44746(process)
 
     return process
```

Reply:
Thanks! I have applied the changes here as well as at PR #44778 in the latest squashed commits.

Reply:
Sorry there seems to be some merge conflicts. I am trying to deal with them

Reply:
The conflicts are solved.